client/post-editor/media-modal/markup.js

```diff
@@ -148,13 +148,19 @@ Markup = {
 				urlOptions = { size: options.size };
 			}
 
-			const img = React.createElement( 'img', {
+			const isTransient = ( 0 === ( media.URL || '' ).indexOf( 'blob:' ) ) || !! media.transient;
```

Comment:
When uploading a local file, the src will be a blob until the upload finishes (then, it is [replaced by URL received from wpcom](https://github.com/Automattic/wp-calypso/blob/master/client/components/tinymce/plugins/media/plugin.jsx#L268)).

As to why the current code checks for a blob, I don't know why that decission was taken, I can only guess: as my understanding goes, every case with `src=blob` should have the transient prop as true; but the transient prop could be true in other cases (upload a file by URL or by external service). Perhaps, in the past, we chose to check for the `src=blob` because that was the only case we wanted to cover?